app/src/main/java/com/sdpteam/connectout/event/EventCreatorActivity.java

```diff
@@ -27,17 +34,33 @@ protected void onCreate(Bundle savedInstanceState) {
         //Upon click on the tool bar icon, end this activity and return to previous one.
         toolbar.setNavigationOnClickListener(v -> this.finish());
 
-        //upon save, log the current event and return to previous activity.
+        //Find given save button.
         Button saveButton = findViewById(R.id.event_creator_save_button);
-        saveButton.setOnClickListener(v ->
-                returnSavedResults());
 
+        //Find given title and description.
+        EditText eventTitle = findViewById(R.id.event_creator_title);
+        EditText eventDescription = findViewById(R.id.event_creator_description);
 
-    }
+        //upon save, log the current event and return to previous activity.
+        saveButton.setOnClickListener(v -> {
+            AuthenticatedUser user = new GoogleAuth().loggedUser();
+            String uid = user == null ? EditProfileActivity.NULL_USER : user.uid;
+
+            //Create associated event.
+            Event newEvent = new Event(
+                    eventTitle.getText().toString(),
+                    new GPSCoordinates(mapFragment.getMovingMarkerPosition()),
+                    eventDescription.getText().toString(),
+                    uid,
+                    UUID.randomUUID().toString()
+            );
+
+            //Save the event & return to previous activity.
+            new EventCreatorViewModel(new EventCreatorModel()).saveValue(newEvent);
+            this.finish();
```

Comment:
This conversation has been resolved IRL, the this.finish() will be preserved for now as it provides the intended behavior of going to the previous activity as it has been agreed on, this might change later on upon linking the activities together & resolution of issues #20 & #21 .

reference:
[How to return to previous activity](https://stackoverflow.com/questions/14966838/how-to-return-to-previous-activity-from-about-with-a-button-click)